### PR TITLE
Fix_React_GraphQL_Mocks

### PR DESCRIPTION
**Pull request by TheNoodleMoose**

### CODE REVIEW

File: src/communityInsights/components/Table/__tests__/Table.tsx [hunk 1]
```diff
@@ -25,34 +25,33 @@ describe('<Table />', () => {
         unmount();
     });
 
-    it('should render properly with data', async () => {
+    fit('should render properly with data', async () => {
```
Comment (OzzieOrca) [inline src/communityInsights/components/Table/__tests__/Table.tsx]: Don't commit `fit` :) I think there may be an eslint rule for this...

File: src/communityInsights/components/Table/__tests__/Table.tsx [hunk 2]
```diff
@@ -25,34 +25,33 @@ describe('<Table />', () => {
...
+                    orgId: '1',
+                },
                 mocks: {
-                    Query: () => ({
-                        globalCommunityChallenges: () => ({
-                            nodes: () => [],
-                            pageInfo: () => {},
+                    GlobalCommunity: () => ({
+                        communityChallenges: () => ({
+                            nodes: () => new MockList(2),
```
Comment (OzzieOrca) [inline src/communityInsights/components/Table/__tests__/Table.tsx]: I think it returns a list of 2 items by default...
https://www.apollographql.com/docs/graphql-tools/mocking/#using-mocklist-in-resolvers
>By default, the summary field will always return 2 entries.

File: src/communityInsights/containers/ChallengesPage/__tests__/ChallengesPage.tsx [hunk 3]
```diff
@@ -15,5 +17,24 @@ describe('<ChallengesPage />', () => {
...
 
-    // TODO: Add a test with mocking data when GraphQL schema contains above query
+    fit('should render properly after loading', async () => {
+        const { snapshot } = renderWithContext(<ChallengesPage />, {
+            appContext: {
+                orgId: '1',
+            },
+            mocks: {
+                Community: () => {
+                    return {
```
Comment (OzzieOrca) [inline src/communityInsights/containers/ChallengesPage/__tests__/ChallengesPage.tsx]: Use the arrow function's default return value?

File: src/communityInsights/components/Table/__tests__/Table.tsx (hunk 1 above)
Comment (TheNoodleMoose) [reply src/communityInsights/components/Table/__tests__/Table.tsx]: Ah good catch! Will remove. Will it make that test the only test that is ran if it is committed?

File: src/communityInsights/containers/ChallengesPage/__tests__/ChallengesPage.tsx (hunk 3 above)
Comment (TheNoodleMoose) [reply src/communityInsights/containers/ChallengesPage/__tests__/ChallengesPage.tsx]: Ah yep, will change!

File: src/communityInsights/components/Table/__tests__/Table.tsx (hunk 1 above)
Comment (OzzieOrca) [reply src/communityInsights/components/Table/__tests__/Table.tsx]: Ya the only test in that file.

File: src/testUtils/globalMocks.ts [hunk 4]
```diff
@@ -14,14 +14,16 @@ export const globalMocks = {
     Float: () => faker.random.number({ precision: 0.01 }),
     Boolean: () => faker.random.boolean(),
     ID: () => nextId(),
+    ISO8601DateTime: () => faker.date.past(10, '2020-01-14').toUTCString(),
+    ISO8601Date: () => faker.date.past(10, '2020-01-14').toUTCString(),
```
Comment (OzzieOrca) [inline src/testUtils/globalMocks.ts]: I think we want this one just to be the date. I was trying to see if we could do it without moment but the answers seemed more complicated.
```suggestion
    ISO8601Date: () => moment(faker.date.past(10, '2020-01-14')).format('YYYY-MM-DD'),
```

**Comment by TheNoodleMoose** [comment]
@OzzieOrca This should be fine to merge in, right?

**Comment by OzzieOrca** [comment]
Yep :)

**Pull request merged**